File: code-examples/scala/indexes/geo.scala
import org.mongodb.scala._
import scala.concurrent.ExecutionContext.Implicits.global
import scala.util.{Failure, Success}

// Replace the placeholders with your MongoDB deployment's credentials
val mongoClient: MongoClient = MongoClient("mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority")
val database = mongoClient.getDatabase("sample_mflix")
val movies = database.getCollection("movies")

val observable = movies.createIndex(Document("location.geo" -> "2dsphere"))

// Running the program and printing the result of the index creation
observable.subscribe(new Observer[String] {
  override def onNext(result: String): Unit = println(s"Index created: $result")
  override def onError(e: Throwable): Unit = println(s"Failed: $e")
  override def onComplete(): Unit = {
    println("Operation Completed")
    mongoClient.close() // Close the connection after the operation completes
  }
})

File: code-examples/php/indexes/geo.php
<?php
require 'vendor/autoload.php';

// Replace placeholders with deployment credentials
$mongoUser = "<user>";
$mongoPwd = "<password>";
$clusterUrl = "<cluster-url>";

// Create MongoDB client
$uri = "mongodb+srv://$mongoUser:$mongoPwd@$clusterUrl";
$client = new MongoDB\Client($uri);

try {
    // Access database and collection
    $database = $client->sample_mflix;
    $movies = $database->movies;

    /* Create a 2dsphere index on the "location.geo" field in the
    "movies" collection */
    $result = $movies->createIndex(["location.geo" => "2dsphere"]);

    // Print the result of the index creation
    echo "Index created: " . $result;

} catch (Exception $e) {
    // Print any thrown errors
    echo 'Caught exception: ',  $e->getMessage(), "\n";

} finally {
    // Close the connection
    $manager = $client->getManager();
    $manager->executeCommand($database, new MongoDB\Driver\Command(['killCursors' => 'movies']));
}
